src/components/public-service-details-page/index.tsx

```diff
@@ -43,17 +68,44 @@ const PublicServiceDetailsPage: FC<Props> = ({
 
     setIsMounted(true);
 
-    return function cleanup() {
+    return () => {
       setIsMounted(false);
+      resetPublicService();
+      resetPublicServices();
     };
-  }, []);
+  }, [publicServiceId]);
```

Comment:
Denne er unødvendig tror jeg

Reply:
Detaljsiden for Tjeneste har relasjoner til Tjeneste (samme detaljside), så hvis id endres så gjøre ny request. Samme useEffect må også gjøre nødvendige resets så ikke det henger igjen. Er det ikke sånn vi må gjøre på alle detaljsider?